robot/connection.py

```diff
@@ -0,0 +1,42 @@
+import json
+import socket
+from typing import Any, Dict
+
+Message = Dict[str, Any]
+
+
+class Connection:
```

Comment:
It feels to me like "codec" would be a much better name for this class and module than "connection", especially since this class doesn't actually do any connecting (that's done in `Board._get_socket`).

Reply:
Possibly, though it also does buffering which I don't think a codec would do. As it stands the name is merely the same as the class this is copied from in [robotd/master.py](https://github.com/sourcebots/robotd/blob/e0be841816d4c778e98498193f6fd7e3de35813d/robotd/master.py#L19-L53). I think it would be good to keep the two classes having the same name, but I don't feel strongly about what the name actually is.

Reply:
I agree that it would be good for the two classes to have the same name; as a result, I think we should rename both of them. Let's leave that to another PR, though.